tools/compute-affected-packages.js

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env node
+
+import fs from "fs";
```

Comment:
can you please add a generic overview of what the script does and how it works?
It would also be a good idea to have a few tests to make sure this works (cant say for sure it works by reading the code)